keras/layers/core.py

```diff
@@ -33,6 +46,25 @@ def __init__(self, **kwargs):
         if not hasattr(self, 'params'):
             self.params = []
 
+    def __call__(self, X):
```

Comment:
@EderSantana , I'm new to keras, but its seems that this logic heavily relies on the order of checks that 'get_input' performs (e.g. check for previous, then for input etc.). Any change in get_input is going to break this logic. It would be nice to have some tests:)